packages/salesforcedx-apex-replay-debugger/test/unit/adapter/apexReplayDebug.test.ts

```diff
@@ -34,8 +34,13 @@ import { HeapDumpService } from '../../../src/core/heapDumpService';
 import { nls } from '../../../src/messages';
 
 export class MockApexReplayDebug extends ApexReplayDebug {
-  public setLogFile(args: LaunchRequestArguments) {
-    this.logContext = new LogContext(args, this);
+  public async setLogFile(args: LaunchRequestArguments): Promise<void> {
+    this.logContext = await LogContext.create(args, this);
+    this.heapDumpService = new HeapDumpService(this.logContext);
+  }
+
+  public async setLogContext(logContext: LogContext): Promise<void> {
```

Comment:
it's a bit silly, but this helps eliminate the linter warning (where we have an `async` but didn't need to.  Here, it has to be async because we're extending a class.  The other option would be to put the eslint comment in to prevent the warning.)

main thing: I'd like to reduce the amount of compiler warnings so that when there is one, it's a real thing to pay attention to.

```suggestion
  public async setLogContext(logContext: LogContext): Promise<void> {
    await Promise.resolve();
```